numpy/core/src/multiarray/mapping.c

```diff
@@ -1383,15 +1383,75 @@ array_subscript_asarray(PyArrayObject *self, PyObject *op)
     return PyArray_EnsureAnyArray(array_subscript(self, op));
 }
 
+/*
+ * Helper function for _get_field_view which turns a multifield
+ * view into a "packed" copy, as done in numpy 1.14 and before.
+ * In numpy 1.15 this function will most likely be removed.
+ */
+NPY_NO_EXPORT int
+_multifield_view_to_copy(PyArrayObject **view) {
+    static PyObject *copyfunc = NULL;
+    PyObject *viewcopy;
+    PyObject *mf_setting = NULL;
+
+    /*
+     * Check "numpy.multifield_view" setting, a secret setting to
+     * help development in the 1.14 to 1.15 transition.
+     * If true, return a view instead of a copy.
+     */
+    mf_setting = npy_import("numpy", "multifield_index_view");
```

Comment:
How do you plan to use this? Are you going to modify `multifield_index_view` on the fly? I'm skeptical of this approach.

Reply:
Yes, it can be modified on the fly, but it is only for testing purposes. See the rewritten section of the docs, and also the unit tests that use it.

Reply:
Still, it introduces unfortunate global state. It would be preferable to have a simple PR that fixes the problem for 1.14.1 and then deal with 1.15.

Reply:
No problem, I'll remove it.